docs(go) rework go guide

gszr opened the pull request.

gszr commented on app/2.1.x/go.md:
````diff
@@ -4,44 +4,128 @@ title: Go language support
+```
+go build -buildmode plugin go-hello.go
+```
+
````
Add an example Dockerfile using steps above to build pluginserver and a plugin

gszr commented on app/2.1.x/go.md:
```diff
@@ -4,44 +4,128 @@ title: Go language support
 
 ## Introduction
 
```
Add minimum Go version required / recommended.

gszr commented on app/2.1.x/go.md:
````diff
@@ -4,44 +4,128 @@ title: Go language support
+go: downloading github.com/ugorji/go/codec v1.1.7
+```
+
+This will, by default, select the latest version available, registering it
````
This paragraph causes the list numbering to be reset.

gszr commented on app/2.1.x/go.md:
````diff
@@ -4,44 +4,128 @@ title: Go language support
+$ go build github.com/Kong/go-pluginserver
+```
+
+**Note**: the following step builds a Go plugin. For the purpose of this tutorial,
````
This paragraph causes the list numbering to be reset.

javierguerragiraldez commented on app/2.1.x/go.md:
```diff
@@ -4,44 +4,128 @@ title: Go language support
+2. The `go-pluginserver` executable. By default, Kong will expect it in
+`/usr/local/bin`; however, this location can be customized with the `go_pluginserver_exe`
+property in the Kong configuration file - or via environment variables, like
+any other Kong configuration property;
```
somewhere we need to say that the go-pluginserver must be compiled.  maybe something like "the `go-pluginserver` executable, built with the same environment as the plugins.  Kong will expect...."

javierguerragiraldez commented on app/2.1.x/go.md:
```diff
@@ -4,44 +4,128 @@ title: Go language support
+in this example, the latest go-pluginserver version available was `v0.5.0` and
+it also pinned the go-pdk version it uses.
+
+3. The Go Plugin Server can be built with the following command, which
```
i feel that using "...can be built..." sounds like there's an option, but it's the only way.  maybe something a bit more imperative, like "3. Build the Go Plugin Server.  The following command results in the executable placed in the current directory.  Move it to where Kong expects it (`/usr/local/bin` by default)"

javierguerragiraldez commented on app/2.1.x/go.md:
```diff
@@ -4,44 +4,128 @@ title: Go language support
+This section shows you how to write a custom Go plugin. It will be useful to
+cross-reference a real Go plugin; see an example [here][go-hello].
+
+### Prerequisites
+
+1. A Kong development environment as usual;
+2. A Go development environment, including a Go compiler;
+3. A `go-pluginserver` executable (check out the previous section to learn how to
+build the Plugin Server).
```
aren't all these already in the previous "prerequisites" section above?

javierguerragiraldez commented on app/2.1.x/go.md:
```diff
@@ -4,44 +4,128 @@ title: Go language support
 for example Go plugins.
 
-### Configuration Structure
+#### Configuration Structure
```
numbers on these subsubsections to match the list?

lena-larionova commented:
Looks like the 2.1.0 branch was created a few days before some other Go PDK doc changes were merged in - this PR is missing content added with https://github.com/Kong/docs.konghq.com/pull/1890.

gszr commented:
> Looks like the 2.1.0 branch was created a few days before some other Go PDK doc changes were merged in - this PR is missing content added with #1890.

Just rebased the release branch onto master, and this branch onto the release branch. I added back the one section with content that is still applicable with our new recommended way of building/using Go plugins.
